### PR TITLE
Pv2 1197 generate levy payments on period end

### PR DESCRIPTION
**Pull request by abroodie**

### CODE REVIEW

File: src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj [hunk 1]
```diff
@@ -90,6 +90,7 @@
     <Compile Include="Properties\AssemblyInfo.cs" />
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (carlbm) [inline src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj]: Can you remove these?

File: src/SFA.DAS.Payments.FundingSource.Application/Repositories/LevyFundingSourceRepository.cs [hunk 2]
```diff
@@ -50,5 +50,22 @@ await dataContext.EmployerProviderPriority
                 .ConfigureAwait(false);
         }
 
+        public async Task<List<long>> GetEmployerAccounts(CancellationToken cancellationToken)
+        {
+            var transferSenders = (await dataContext.Apprenticeship
+                .Where(apprenticeship => apprenticeship.TransferSendingEmployerAccountId != null)
```
Comment (carlbm) [inline src/SFA.DAS.Payments.FundingSource.Application/Repositories/LevyFundingSourceRepository.cs]: Elsewhere this was changed to != null and != 0. Should that change be replicated here?

File: src/SFA.DAS.Payments.JobContextMessageHandling/JobContextManagerService.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            this.jobContextManager = jobContextManager ?? throw new ArgumentNullException(nameof(jobContextManager));
+        }
+
+        public async Task RunAsync(CancellationToken cancellationToken)
+        {
+            var initialised = false;
+            try
+            {
+                logger.LogDebug("Starting the Earning Events service.");
+                jobContextManager.OpenAsync(cancellationToken);
```
Comment (carlbm) [inline src/SFA.DAS.Payments.JobContextMessageHandling/JobContextManagerService.cs]: is this missing an await?

File: src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj (hunk 1 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj]: done

File: src/SFA.DAS.Payments.FundingSource.Application/Repositories/LevyFundingSourceRepository.cs (hunk 2 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.FundingSource.Application/Repositories/LevyFundingSourceRepository.cs]: done

File: src/SFA.DAS.Payments.JobContextMessageHandling/JobContextManagerService.cs (hunk 3 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.JobContextMessageHandling/JobContextManagerService.cs]: nope, this is an incorrectly named sync method

File: src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj]: removed

File: src/SFA.DAS.Payments.AcceptanceTests.Components.Core/app.config [hunk 4]
```diff
@@ -0,0 +1,55 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.Components.Core/app.config]: removed

File: src/SFA.DAS.Payments.AcceptanceTests.Core/SFA.DAS.Payments.AcceptanceTests.Core.csproj [hunk 5]
```diff
@@ -416,6 +419,7 @@
     <Compile Include="Services\UkprnService.cs" />
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
[the remark this answers is not in the document]
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.Core/SFA.DAS.Payments.AcceptanceTests.Core.csproj]: removed

File: src/SFA.DAS.Payments.AcceptanceTests.Core/app.config [hunk 6]
```diff
@@ -0,0 +1,55 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.Core/app.config]: removed

File: src/SFA.DAS.Payments.AcceptanceTests.Core/packages.config [hunk 7]
```diff
@@ -15,7 +15,8 @@
   <package id="ESFA.DC.JobContextManager" version="0.2.5-beta0001" targetFramework="net471" />
   <package id="ESFA.DC.Jobs.Model" version="0.1.28-beta0001" targetFramework="net471" />
   <package id="ESFA.DC.JobStatus.Interface" version="0.2.5-beta0001" targetFramework="net471" />
-  <package id="ESFA.DC.Logging" version="0.1.1-beta0001" targetFramework="net471" />
+  <package id="ESFA.DC.Logging" version="0.1.4-beta0001" targetFramework="net471" />
```
[the remark this answers is not in the document]
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.Core/packages.config]: fixed

File: src/SFA.DAS.Payments.Database/Payments2/Tables/Payments2.Apprenticeship.sql [hunk 8]
```diff
@@ -20,4 +20,14 @@
     [IsLevyPayer] BIT NOT NULL,
 	CreationDate DATETIMEOFFSET NOT NULL CONSTRAINT DF_Apprenticeship__CreationDate DEFAULT (SYSDATETIMEOFFSET()),
 
-)
\ No newline at end of file
+)
+GO
+
+CREATE INDEX [IX_Apprenticeship__AccountSearch] ON [Payments2].[Apprenticeship]
```
[the remark this answers is not in the document]
Comment (abroodie) [reply src/SFA.DAS.Payments.Database/Payments2/Tables/Payments2.Apprenticeship.sql]: DACPAC script will check this

File: src/SFA.DAS.Payments.JobContextMessageHandling/JobContextManagerService.cs (hunk 3 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.JobContextMessageHandling/JobContextManagerService.cs]: 😂

**Pull request merged**